internal/identity_vm.go

```diff
@@ -131,5 +131,5 @@ func fullyQualifiedAppID(_ netcontext.Context) string {
 }
 
 func IsDevAppServer() bool {
-	return os.Getenv("RUN_WITH_DEVAPPSERVER") != ""
+	return os.Getenv("RUN_WITH_DEVAPPSERVER") != "" || os.Getenv("GAE_ENV") == "localdev"
```

Comment:
FYI: in the case where these two variable disagree (e.g. `RUN_WITH_DEVAPPSERVER=1 GAE_ENV=standard`) this code trusts the legacy RUN_WITH_DEVAPPSERVER over the new GAE_ENV.  IMO this is the best option as it is less likely to break existing test environments that may be configured this way, and lets people start using the new variable in those envs.  We can update this logic and/or totally remove RUN_WITH_DEVAPPSERVER in a later release.